src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -349,18 +349,20 @@ void prepare_primitive_fusing::fuse_bias(program &p) {
         if (node->get_output_layout().is_dynamic())
             continue;
 
-        size_t out_features = static_cast<size_t>(node->get_output_layout().feature());
+        cldnn::tensor::value_type out_features = node->get_output_layout().feature();
 
         // Change out_features value to proper dimension for 3D FC case
         if (is_3d_fully_connected(node->get_dependency(0)))
-            out_features = static_cast<size_t>(node->get_dependency(0).get_output_layout().spatial(1));
+            out_features = node->get_dependency(0).get_output_layout().spatial(1);
         else if (is_3d_fully_connected(node->get_dependency(1)))
-            out_features = static_cast<size_t>(node->get_dependency(1).get_output_layout().spatial(1));
+            out_features = node->get_dependency(1).get_output_layout().spatial(1);
 
         int bias_idx = -1;
         for (size_t i = 0; i < eltw_node.get_dependencies().size(); i++) {
             auto& dep = eltw_node.get_dependency(i);
-            if (dep.is_constant() && dep.get_output_layout().count() == out_features) {
+            if (dep.is_constant() &&
+                dep.get_output_layout().feature() == out_features &&
+                dep.get_output_layout().count() == static_cast<size_t>(out_features)) {
```

Comment:
If `dep.get_output_layout().feature()` is same with `out_fetureas` and `dep.get_output_layout().count()` is same with `out_features`, `dep.get_output_layout().feature()` and `dep.get_output_layout().count()` should be always same. 
Is this checking condition only allowing the case of that the output layout has only feature value?

Reply:
Single `dep.get_output_layout().feature() == dep.get_output_layout().count()` check won't be safe as in cases when `out_feature != dep.get_output_layout().feature()` we'll get either exception or UB.